### PR TITLE
enhancement: #500 Flag for the ScrollSpy component that ensures the last active element remains visible + animationDuration option

### PR DESCRIPTION
**Pull request by IvanPostu**
fixes https://github.com/materializecss/materialize/issues/500

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist:

- [x] I have read the **[CONTRIBUTING document](https://github.com/materializecss/materialize/blob/main/CONTRIBUTING.md)**.
- [x] My commit messages follows the [conventional commit format](https://github.com/materializecss/materialize/blob/main/CONTRIBUTING.md#submitting-your-pull-request)
- [x] My change requires a change to the documentation, and updated it accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by wuda-io** [comment]
Nice, your video explaines it very clearly 👍 i will review the code tomorrow.

File: src/scrollspy.ts [hunk 1]
```diff
@@ -250,3 +301,30 @@ export class ScrollSpy extends Component<ScrollSpyOptions> {
...
+
+    if (progress < 1) {
+      window.scrollTo(0, scrollY);
+      requestAnimationFrame(scrollStep);
+    } else {
+      window.scrollTo(0, targetPosition);
+    }
+  }
+  requestAnimationFrame(scrollStep);
+}
```
Comment (wuda-io) [inline src/scrollspy.ts:330]: i think this two functions should be inside of the class. The rest looks nice!
Can you place them as private functions, then i will merge. Thanks

Comment (IvanPostu) [reply src/scrollspy.ts]: :heavy_check_mark:

**Review by wuda-io — APPROVED** [review]

**Comment by wuda-io** [comment]
Please use more descriptive commit messages.
Also apply commit lint format so we can properly generate the changelog.

**Comment by wuda-io** [comment]
Could you please also update the docs?
Please make a PR here https://github.com/materializecss/materialize-docs
Make sure  you use the dev branch.

**Pull request merged**